### ISSUE TITLE
Terribly sorry I was trying to contact you about an old Dec Alpha Qemu project you had done

### ISSUE THREAD

**Issue by neozeed** [tgingold-cern/cheby#3]
While looking up ARC on the Alpha I came across a few patches you had made a LONG time ago, and managed to find this repo:

https://repo.or.cz/w/qemu/es40.git

You had mentioned that you had been able to get into AlphaBIOS/Arc but I was unable to reproduce with this branch.

I don't suppose you would be able to reproduce getting into ARC?

With the recent surfacing of a 64bit version of Windows 2000 for the Alpha there is a resurgent interest in Alpha's but of coursing finding hardware is so incredibly rare.

I had covered it here:

https://virtuallyfun.com/2023/05/15/windows-2000-64-bit-for-alpha-axp/

Thanks

**Comment by tgingold-cern** [comment]
Hello,

I reply to both of you as I got your messages mostly at the same time.

So yes, I was able to load ARCBios, at least to the point I got the arc splash screen and I think a prompt.

I think the repo contains all the code I have worked on.  Also, I was more interested in the pal code for VMS than the pal code for Windows.
So maybe ARC worked with some older commits.  I also remember that I was not able to boot NT, but I don't remember whether it was due to a crash in ARC or in NT.

I know for sure that some standard alpha instructions have been fixed by R. Henderson after my work.

I haven't followed Qemu for a long time, so I have no idea where you should start (from scratch or from my work).

I think all the Alpha cpu and PAL were documented deeply enough to fully emulate it.

Tell me if you make progress!

Cheers,
Tristan


________________________________
From: neozeed ***@***.***>
Sent: Tuesday, June 13, 2023 12:19 PM
To: tgingold-cern/cheby ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [tgingold-cern/cheby] Terribly sorry I was trying to contact you about an old Dec Alpha Qemu project you had done (Issue #3)


While looking up ARC on the Alpha I came across a few patches you had made a LONG time ago, and managed to find this repo:

https://repo.or.cz/w/qemu/es40.git

You had mentioned that you had been able to get into AlphaBIOS/Arc but I was unable to reproduce with this branch.

I don't suppose you would be able to reproduce getting into ARC?

With the recent surfacing of a 64bit version of Windows 2000 for the Alpha there is a resurgent interest in Alpha's but of coursing finding hardware is so incredibly rare.

I had covered it here:

https://virtuallyfun.com/2023/05/15/windows-2000-64-bit-for-alpha-axp/

Thanks

**Comment by neozeed** [comment]
![alphabios](https://github.com/tgingold-cern/cheby/assets/9031439/b71ac1f9-6dee-4eab-a5c9-78e4b36e2537)
It took me a while to figure it out, as I always had a suspicion the SRM didn't contain the Alpha BIOS.  

I found some page with instructions on updating an Alpha's flash so that's what I did, I fully programmed a 2MB flash file, and yeah it'll boot into ARC!

I can't believe that emulating AlphaNT was so far along.  I've just spent the day reproducing your work, so thanks again.  I can't promise anything but maybe it'll inspire others!

[qemu-es40-alphabios.zip](https://github.com/tgingold-cern/cheby/files/11746776/qemu-es40-alphabios.zip)
